### PR TITLE
Update zif_abap_serverless_v1.intf.abap

### PR DESCRIPTION
**Pull request by larshp**
change to `body`, as its called body

### CODE REVIEW

**Comment by larshp** [comment]
well, body should actually be hex, but its easier to start out with strings

**Comment by mrsimpson** [comment]
I’m fine with this change. However, I think we should add more things to the interface, such as `path` and `query`. 
How do we like to organize? Merge everything with it’s fine or find „bigger solutions“? I’m fine with either, feel free to merge!

**Comment by larshp** [comment]
path and query is part of the http headers, this is a raw approach, where the surrounding framework does not do much

instead logic is pushed to the ABAP part, which is a great language to do stuff in 😄 we can build different frameworks for parsing the headers and body in ABAP

**Comment by mrsimpson** [comment]
> path and query is part of the http headers, this is a raw approach, where the surrounding framework does not do much

I have to disagree: path and query are separate locations and outside the http header.

See also the [OAS3 spec concerning parameters](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameterObject) 

```
Parameter Locations
There are four possible parameter locations specified by the in field:

path - Used together with Path Templating, where the parameter value is actually part of the operation's URL. This does not include the host or base path of the API. For example, in /items/{itemId}, the path parameter is itemId.
query - Parameters that are appended to the URL. For example, in /items?id=###, the query parameter is id.
header - Custom headers that are expected as part of the request. Note that RFC7230 states header names are case insensitive.
cookie - Used to pass a specific cookie value to the API.
```

...or dev console of your browser 😁

**Comment by larshp** [comment]
hmm

**Comment by larshp** [comment]
http is https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html

so adding a `method` and `request_uri` as inputs should do it? ignoring version, remaining is header fields?

**Comment by mrsimpson** [comment]
In "raw" http holds path and query as part of the uri - not inside the header, but as a concatenated string.
Since you as a provider anyway consume this information separately, I would prefer to parse it outside (as part of the framework), see https://www.npmjs.com/package/url (and its various adaptations, such as in express.js)

File: src/zif_abap_serverless_v1.intf.abap [hunk 1]
```diff
@@ -15,9 +15,11 @@ INTERFACE zif_abap_serverless_v1 PUBLIC.
   METHODS
     run
       IMPORTING
-        is_request TYPE ty_http
+        iv_method      TYPE string
+        iv_request_uri TYPE string
```
Comment (mrsimpson) [inline src/zif_abap_serverless_v1.intf.abap]: Better, but most consumers will already feature a parsed uri (`path` and `query`). Are you so reluctant to use that?

Comment (larshp) [reply src/zif_abap_serverless_v1.intf.abap]: in OpenFaaS is it parsed as part of the protocol specification?

Comment (mrsimpson) [reply src/zif_abap_serverless_v1.intf.abap]: In express (which I use as http-framework in the OpenFaaS node image) it is

**Comment by larshp** [comment]
updated, now it looks like https://github.com/openfaas/templates/blob/master/template/java8/model/src/main/java/com/openfaas/model/IRequest.java without the helper methods

**Comment by mrsimpson** [comment]
hm - not really: query is an enumerable in most of the cases https://github.com/openfaas/templates/blob/master/template/java8/model/src/main/java/com/openfaas/model/IRequest.java#L13
but that would raise the question of the line type...

**Comment by larshp** [comment]
yea, thats a helper method

what if the query string is malformed? then the ABAP code should not be called, or without the real inputs, or should there like the java example be 2 ways to get the same data? should there then be a flag to tell if the query was parsed correctly by the surrounding framework?

If supplying a table, then the framework need to supply these values, which is difficult as its a table, this gives a larger face to the runtime, making it more difficult to change the runtime. 

Putting the responsibility to ABAP gives full flexibility and transparency on ABAP side. If there is an error parsing the query on ABAP side, the ABAP developer can fix it. If its part of the framework, the developer cannot fix it right away.

**Comment by larshp** [comment]
the ABAP developer can create an abstract class, implementing the interface, exposing the helper methods to the subclasses

**Comment by mrsimpson** [comment]
> Putting the responsibility to ABAP gives full flexibility and transparency on ABAP side. If there is an error parsing the query on ABAP side, the ABAP developer can fix it. If its part of the framework, the developer cannot fix it right away.

Go for the string

**Comment by larshp** [comment]
If there are are 5 serverless server implementations, and the parsing is on the ABAP side, it will use the same logic always, guaranteed. Instead of 5 different surrounding implementations, which potentially can have different implementations.

Parsing query and path should be an easy task, but anyhow 🔨

**Comment by mrsimpson** [comment]
Worst thing to happen is serializing an parsing it once more

**Pull request merged**